<a href="https://colab.research.google.com/github/RodrigoFigueroa19/dojo_proyecto1/blob/main/predicciones_de_ventas_de_alimentos.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [24]:


import pandas as pd

filename = "/content/drive/MyDrive/Dojo/Unidad 1/Proyecto1/sales_predictions_2023.csv"
# importamos Pandas
df = pd.read_csv(filename) # Carga de Archivo

df



Item_Identifier  Item_Weight Item_Fat_Content  Item_Visibility  \
0              FDA15        9.300          Low Fat         0.016047   
1              DRC01        5.920          Regular         0.019278   
2              FDN15       17.500          Low Fat         0.016760   
3              FDX07       19.200          Regular         0.000000   
4              NCD19        8.930          Low Fat         0.000000   
...              ...          ...              ...              ...   
8518           FDF22        6.865          Low Fat         0.056783   
8519           FDS36        8.380          Regular         0.046982   
8520           NCJ29       10.600          Low Fat         0.035186   
8521           FDN46        7.210          Regular         0.145221   
8522           DRG01       14.800          Low Fat         0.044878   

                  Item_Type  Item_MRP Outlet_Identifier  \
0                     Dairy  249.8092            OUT049   
1               Soft Drinks   48.2692            OUT018   
2                      Meat  141.6180            OUT049   
3     Fruits and Vegetables  182.0950            OUT010   
4                 Household   53.8614            OUT013   
...                     ...       ...               ...   
8518            Snack Foods  214.5218            OUT013   
8519           Baking Goods  108.1570            OUT045   
8520     Health and Hygiene   85.1224            OUT035   
8521            Snack Foods  103.1332            OUT018   
8522            Soft Drinks   75.4670            OUT046   

      Outlet_Establishment_Year Outlet_Size Outlet_Location_Type  \
0                          1999      Medium               Tier 1   
1                          2009      Medium               Tier 3   
2                          1999      Medium               Tier 1   
3                          1998         NaN               Tier 3   
4                          1987        High               Tier 3   
...                         ...         ...                  ...   
8518                       1987        High               Tier 3   
8519                       2002         NaN               Tier 2   
8520                       2004       Small               Tier 2   
8521                       2009      Medium               Tier 3   
8522                       1997       Small               Tier 1   

            Outlet_Type  Item_Outlet_Sales  
0     Supermarket Type1          3735.1380  
1     Supermarket Type2           443.4228  
2     Supermarket Type1          2097.2700  
3         Grocery Store           732.3800  
4     Supermarket Type1           994.7052  
...                 ...                ...  
8518  Supermarket Type1          2778.3834  
8519  Supermarket Type1           549.2850  
8520  Supermarket Type1          1193.1136  
8521  Supermarket Type2          1845.5976  
8522  Supermarket Type1           765.6700  

[8523 rows x 12 columns]

In [25]:
len (df.columns )

12

Proyecto parte 2
R1 .- Tenemos 12 Columnas


In [26]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8523 entries, 0 to 8522
Data columns (total 12 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Item_Identifier            8523 non-null   object 
 1   Item_Weight                7060 non-null   float64
 2   Item_Fat_Content           8523 non-null   object 
 3   Item_Visibility            8523 non-null   float64
 4   Item_Type                  8523 non-null   object 
 5   Item_MRP                   8523 non-null   float64
 6   Outlet_Identifier          8523 non-null   object 
 7   Outlet_Establishment_Year  8523 non-null   int64  
 8   Outlet_Size                6113 non-null   object 
 9   Outlet_Location_Type       8523 non-null   object 
 10  Outlet_Type                8523 non-null   object 
 11  Item_Outlet_Sales          8523 non-null   float64
dtypes: float64(4), int64(1), object(7)
memory usage: 799.2+ KB


In [ ]:
Aca el detalle de cada Variable y tenemos 8523 Filas



In [28]:
df.duplicated()

0       False
1       False
2       False
3       False
4       False
        ...  
8518    False
8519    False
8520    False
8521    False
8522    False
Length: 8523, dtype: bool

In [29]:
df.duplicated().sum()

0

No hay Duplicados


In [30]:
df["Outlet_Type"].value_counts()
df.isna().sum()




Item_Identifier                 0
Item_Weight                  1463
Item_Fat_Content                0
Item_Visibility                 0
Item_Type                       0
Item_MRP                        0
Outlet_Identifier               0
Outlet_Establishment_Year       0
Outlet_Size                  2410
Outlet_Location_Type            0
Outlet_Type                     0
Item_Outlet_Sales               0
dtype: int64

Faltan DAtos en Outlet_size 2410 e Item_weight son 1463


In [31]:
df = df.drop(columns=['Item_Weight', 'Outlet_Size'])
df


Item_Identifier Item_Fat_Content  Item_Visibility              Item_Type  \
0              FDA15          Low Fat         0.016047                  Dairy   
1              DRC01          Regular         0.019278            Soft Drinks   
2              FDN15          Low Fat         0.016760                   Meat   
3              FDX07          Regular         0.000000  Fruits and Vegetables   
4              NCD19          Low Fat         0.000000              Household   
...              ...              ...              ...                    ...   
8518           FDF22          Low Fat         0.056783            Snack Foods   
8519           FDS36          Regular         0.046982           Baking Goods   
8520           NCJ29          Low Fat         0.035186     Health and Hygiene   
8521           FDN46          Regular         0.145221            Snack Foods   
8522           DRG01          Low Fat         0.044878            Soft Drinks   

      Item_MRP Outlet_Identifier  Outlet_Establishment_Year  \
0     249.8092            OUT049                       1999   
1      48.2692            OUT018                       2009   
2     141.6180            OUT049                       1999   
3     182.0950            OUT010                       1998   
4      53.8614            OUT013                       1987   
...        ...               ...                        ...   
8518  214.5218            OUT013                       1987   
8519  108.1570            OUT045                       2002   
8520   85.1224            OUT035                       2004   
8521  103.1332            OUT018                       2009   
8522   75.4670            OUT046                       1997   

     Outlet_Location_Type        Outlet_Type  Item_Outlet_Sales  
0                  Tier 1  Supermarket Type1          3735.1380  
1                  Tier 3  Supermarket Type2           443.4228  
2                  Tier 1  Supermarket Type1          2097.2700  
3                  Tier 3      Grocery Store           732.3800  
4                  Tier 3  Supermarket Type1           994.7052  
...                   ...                ...                ...  
8518               Tier 3  Supermarket Type1          2778.3834  
8519               Tier 2  Supermarket Type1           549.2850  
8520               Tier 2  Supermarket Type1          1193.1136  
8521               Tier 3  Supermarket Type2          1845.5976  
8522               Tier 1  Supermarket Type1           765.6700  

[8523 rows x 10 columns]

Se eliminan las columas con los datos faltantes en lugar del tamaño de la tienda usaremos el tipo de tienda y el peso del item sera descartado del analisis


In [32]:
df.isna().sum()

Item_Identifier              0
Item_Fat_Content             0
Item_Visibility              0
Item_Type                    0
Item_MRP                     0
Outlet_Identifier            0
Outlet_Establishment_Year    0
Outlet_Location_Type         0
Outlet_Type                  0
Item_Outlet_Sales            0
dtype: int64

No tenemos Datos Nulos

In [37]:
import numpy as np
columnas = np.array (df.columns.values)

len (columnas)

for i in range  (0,len(columnas) - 1,1) :

      print (df[columnas[i ]].value_counts())


Item_Identifier
FDW13    10
FDG33    10
NCY18     9
FDD38     9
DRE49     9
         ..
FDY43     1
FDQ60     1
FDO33     1
DRF48     1
FDC23     1
Name: count, Length: 1559, dtype: int64
Item_Fat_Content
Low Fat    5089
Regular    2889
LF          316
reg         117
low fat     112
Name: count, dtype: int64
Item_Visibility
0.000000    526
0.076975      3
0.162462      2
0.076841      2
0.073562      2
           ... 
0.013957      1
0.110460      1
0.124646      1
0.054142      1
0.044878      1
Name: count, Length: 7880, dtype: int64
Item_Type
Fruits and Vegetables    1232
Snack Foods              1200
Household                 910
Frozen Foods              856
Dairy                     682
Canned                    649
Baking Goods              648
Health and Hygiene        520
Soft Drinks               445
Meat                      425
Breads                    251
Hard Drinks               214
Others                    169
Starchy Foods             148
Breakfast                 1

Item_Fat_Content tiene categorias extrañas para Low Fat



In [38]:
df["Item_Fat_Content"].value_counts()



Item_Fat_Content
Low Fat    5089
Regular    2889
LF          316
reg         117
low fat     112
Name: count, dtype: int64

In [39]:


df['Item_Fat_Content'] = df['Item_Fat_Content'].replace('LF', 'Low Fat') # indicamos que lf es Low Fat
df['Item_Fat_Content'] = df['Item_Fat_Content'].replace('reg', 'Regular') # indicamos que reg es Regular
df['Item_Fat_Content'] = df['Item_Fat_Content'].replace('low fat', 'Low Fat') # indicamos que reg es Regular


df['Item_Fat_Content'].value_counts()

Item_Fat_Content
Low Fat    5517
Regular    3006
Name: count, dtype: int64

Corregimos el Item de contenido de Grasa



In [40]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8523 entries, 0 to 8522
Data columns (total 10 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Item_Identifier            8523 non-null   object 
 1   Item_Fat_Content           8523 non-null   object 
 2   Item_Visibility            8523 non-null   float64
 3   Item_Type                  8523 non-null   object 
 4   Item_MRP                   8523 non-null   float64
 5   Outlet_Identifier          8523 non-null   object 
 6   Outlet_Establishment_Year  8523 non-null   int64  
 7   Outlet_Location_Type       8523 non-null   object 
 8   Outlet_Type                8523 non-null   object 
 9   Item_Outlet_Sales          8523 non-null   float64
dtypes: float64(3), int64(1), object(6)
memory usage: 666.0+ KB


In [45]:
df["Item_Visibility"].max(),df["Item_Visibility"].min(),df["Item_Visibility"].sum()


(0.328390948, 0.0, 563.6432812830001)

In [46]:
df["Item_MRP"].max(),df["Item_MRP"].min(),df["Item_MRP"].sum()

(266.8884, 31.29, 1201681.4808)

In [47]:
df["Outlet_Establishment_Year"].max(),df["Outlet_Establishment_Year"].min(),df["Outlet_Establishment_Year"].sum()

(2009, 1985, 17027521)

In [48]:
df["Item_Outlet_Sales"].max(),df["Item_Outlet_Sales"].min(),df["Item_Outlet_Sales"].sum()

(13086.9648, 33.29, 18591125.4104)